extensions/wikia/TemplateDraft/ApprovedraftAction.php

```diff
@@ -70,12 +70,15 @@ private function approveDraft( Title $draftTitle ) {
 		global $wgEnableInsightsInfoboxes;
 
 		// Get Title object of parent page
-		$helper = new TemplateDraftHelper();
-		$parentTitle = $helper->getParentTitle( $draftTitle );
+		$parentTitle = TemplateDraftHelper::getParentTitle( $draftTitle );
 
 		// Check edit rights
-		if ( !$parentTitle->userCan( 'templatedraft' ) ) {
-			throw new PermissionsException( 'edit' );
+		if ( !$parentTitle->userCan( 'templatedraft' ) || !$parentTitle->userCan( 'edit' ) ) {
+			throw new ErrorPageError( 'badaccess', 'templatedraft-protect-edit' );
+		}
+
+		if ( !$parentTitle->userCan( 'move' ) ) {
```

Comment:
yes, if a template is only move protected, a non-admin should be able to approve.

However, if this is very complicated to solve for, we should open another ticket and solve it later (if it ever becomes a real-world problem for users.)